### ISSUE TITLE
Almost all mods in my instances are marked as outdated, but they are up-to-date

### ISSUE THREAD

**Issue by 66BA9Q-ME49** [Voxelum/x-minecraft-launcher#820]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Launcher Version

0.47.8

### Operating System

Windows

### Architecture

x64

### Current Behavior

When I checked for mod updates, almost every mod are flagged as outdated, but they're on the latest version.

### Expected Behavior

Only outdated mods should be marked as outdated.

### Steps To Reproduce

1. Check for mod updates in an instance
2. A red dot appears for almost every mods

### Anything else?

![image](https://github.com/user-attachments/assets/334c8113-5634-43f7-a560-5fd62b71068b)

I freshly installed all these mods on my bug reporting instance, and they are flagged as outdated.

**Comment by ci010** [comment]
> The update checker is now working correctly, but all the update markers are still disappearing each time I update a mod.

yes, the update scheduale will reset after the mods change. I may use smarter algorithm here to keep the un-updated files
